Shared/Samples/Generate offline map with custom parameters/GenerateOfflineMapWithCustomParametersView.swift

```diff
@@ -0,0 +1,434 @@
+                  let extent = parameters.areaOfInterest?.extent else { return }
+            
+            // Creates the generate offline map job based on the parameters and overrides.
+            generateOfflineMapJob = offlineMapTask.makeGenerateOfflineMapJob(
+                parameters: parameters,
+                downloadDirectory: temporaryDirectory,
+                overrides: overrides
+            )
+            
+            // Starts the job.
+            generateOfflineMapJob.start()
+            
+            defer {
+                generateOfflineMapJob = nil
+                isGenerateDisabled = offlineMap != nil
+            }
+            
+            // Awaits the output of the job.
+            let output = try await generateOfflineMapJob.output
+            // Sets the offline map to the output's offline map.
+            offlineMap = output.offlineMap
+            // Sets the initial viewpoint of the offline map.
+            offlineMap.initialViewpoint = Viewpoint(boundingGeometry: extent.expanded(by: 0.8))
+        }
+        
+        /// Cancels the generate offline map job.
+        func cancelJob() async {
+            // Cancels the generate offline map job.
+            await generateOfflineMapJob?.cancel()
+            generateOfflineMapJob = nil
+        }
+        
+        /// Creates a temporary directory.
+        private static func createTemporaryDirectory() -> URL {
+            // swiftlint:disable:next force_try
+            try! FileManager.default.url(
+                for: .itemReplacementDirectory,
+                in: .userDomainMask,
+                appropriateFor: FileManager.default.temporaryDirectory,
+                create: true
+            )
+        }
+        
+        // MARK: - Basemap helpers
+        private func getExportTileCacheParametersForBasemapLayer() -> ExportTileCacheParameters? {
+            if let basemapLayer = onlineMap.basemap?.baseLayers.first as? Layer {
+                if let key = OfflineMapParametersKey(layer: basemapLayer) {
+                    return offlineMapParameterOverrides?.exportTileCacheParameters[key]
+                }
+            }
+            return nil
+        }
+        
+        /// Sets the scale level range so that only the levels between the min and max inclusive,
+        /// are downloaded. Note that lower values are zoomed further out,
+        /// i.e. 0 has the least detail, but one tile covers the entire Earth.
+        /// Parameters:
+        /// - minScaleLevel: The minumum scale level to download foe the basemap.
+        /// - maxScaleLevel: The maximun scale level to download for the basemap.
+        func restrictBasemapScaleLevelRangeTo(minScaleLevel: Double, maxScaleLevel: Double) {
+            guard let tileCacheParameters = getExportTileCacheParametersForBasemapLayer(),
+                  // Ensure that the lower bound of the range is not greater than the upper bound
```

Comment:
```suggestion
                  // Ensure that the lower bound of the range is not greater than the upper bound.
```
Comment trailing comma, same for other comments below.

Reply:
I think I got them all.